packages/core/admin/admin/src/pages/SettingsPage/components/Tokens/Table/DeleteButton/tests/DeleteButton.test.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { ThemeProvider, lightTheme } from '@strapi/design-system';
+import { act, fireEvent, render } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
+import { NotificationsProvider } from '@strapi/helper-plugin';
+
+import DeleteButton from '../index';
+
+jest.mock('@strapi/helper-plugin', () => ({
+  ...jest.requireActual('@strapi/helper-plugin'),
+  useTracking: jest.fn(() => ({ trackUsage: jest.fn() })),
+}));
+
+function getButton(container, name) {
+  return container.querySelector(`button[name="${name}"]`);
+}
+
+function ComponentToTest(props) {
+  return (
+    <IntlProvider locale="en" messages={{}}>
+      <ThemeProvider theme={lightTheme}>
+        <NotificationsProvider toggleNotification={() => {}}>
+          <DeleteButton tokenName="test" onClickDelete={() => {}} {...props} />
+        </NotificationsProvider>
+      </ThemeProvider>
+    </IntlProvider>
+  );
+}
+
+const setup = (props = { onClickDelete: jest.fn() }) => {
+  return render(<ComponentToTest {...props} />);
+};
+
+describe('DeleteButton', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  it('renders and matches the snapshot', () => {
+    const { baseElement } = setup();
+    expect(baseElement).toMatchSnapshot();
+  });
+  it('show confirmation delete dialog when the delete button is clicked', () => {
+    const { baseElement, queryByText } = setup();
+    act(() => {
+      fireEvent.click(getButton(baseElement, 'delete'));
+    });
+
+    expect(queryByText('Are you sure you want to delete this?')).toBeInTheDocument();
+  });
+
+  it('closes the modal when you click on Cancel button', () => {
+    const { baseElement, queryByText, getByText } = setup();
+    act(() => {
+      fireEvent.click(getButton(baseElement, 'delete'));
+    });
```

Comment:
is this `act` necessary? It doesn't perform an asynchronous update.

Reply:
removed